File: tasks/2022/qualification/isogram.py
'''
Adott sor kvazi-izogram, ha a benne leggyakrabban elofordulo betu gyakorisaga legfeljebb 2 (vagy egyaltalan nem fordult benne elo betu).
A gyakorisagok szamontartasahoz hasznalhatunk peldaul tombot vagy szotarat (C++ nyelven std::map vagy std::unordered_map).
Fontos, hogy csak a betukkel foglalkozzunk, az egyeb karaktereket hagyjuk figyelmen kivul.
'''

from collections import defaultdict

n = int(input())

def quasi_isogram(line):
    chars = defaultdict(lambda: 0)
    for c in filter(lambda c: c.isalpha(), line.lower()):
        chars[c] += 1
    return 1 if len(chars) == 0 or max(chars.values()) <= 2 else 0

print(sum([quasi_isogram(input()) for _ in range(n)]))

File: tasks/2022/qualification/isogram.cpp
/*
Adott sor kvázi-izogram, ha bármely benne előforduló betű gyakorisága legfeljebb 2.
A gyakoriságok számontartásához használhatunk például tömböt vagy szótárat (std::map vagy std::unordered_map).
Fontos, hogy csak a betűkkel foglalkozzunk, az egyéb karaktereket hagyjuk figyelmen kívül.
*/

#include <bits/stdc++.h>
using namespace std;
int n, ans;
string tmp;

bool quasi_isogram(string &tmp)
{
    int cnt[26] = {0};
    for (char c: tmp)
    {
        if (c >= 'a' && c <= 'z') cnt[c - 'a']++;
        if (c >= 'A' && c <= 'Z') cnt[c - 'A']++;
    }
    for (int i = 0; i < 26; ++i)
    {
        if (cnt[i] > 2) return false;
    }
    return true;
}

int main()
{
    cin >> n;
    getline(cin, tmp);
    for (int i = 0; i < n; ++i)
    {
        getline(cin, tmp);
        if (quasi_isogram(tmp)) ans++;
    }
    cout << ans << "\n";
}